HeliPort/Appearance/StatusMenu.swift

```diff
@@ -43,6 +46,11 @@ final class StatusMenu: NSMenu, NSMenuDelegate {
 
             statusItem.title = NSLocalizedString(status.description)
 
+            if status != ITL80211_S_RUN && previousStatus == ITL80211_S_RUN && !isDisconnecting {
```

Comment:
The conditions here are not robust enough. For example, in https://github.com/OpenIntelWireless/itlwm/issues/934#issuecomment-2001892269 the router requests a reassociation. If we caught this event, we might mistakenly disconnect from the current network.

Many other factors might cause a change in state as well, such as firmware crash (goes directly back to `S_INIT`, no intervention from HeliPort required, itlwm will recover by itself), key expiring (-> `S_AUTH`), etc. The idea is to have our own state adapt and tolerate well with itlwm's internal state.